beacon_node/http_api/src/block_id.rs

```diff
@@ -266,6 +267,27 @@ impl BlockId {
             Err(e) => Err(warp_utils::reject::beacon_chain_error(e)),
         }
     }
+
+    pub async fn blob_sidecar_list_filtered<T: BeaconChainTypes>(
+        &self,
+        indices: BlobIndicesQuery,
+        chain: &BeaconChain<T>,
+    ) -> Result<BlobSidecarList<T::EthSpec>, warp::Rejection> {
+        let blob_sidecar_list = self.blob_sidecar_list(&chain).await?;
+        let blob_sidecar_list_filtered = match indices.indices {
+            Some(vec) => {
+                let list = blob_sidecar_list
+                    .into_iter()
+                    .filter(|blob_sidecar| vec.contains(&blob_sidecar.index))
+                    .collect();
+                let filtered_list = BlobSidecarList::new(list)
+                    .map_err(|e| warp_utils::reject::custom_server_error(format!("{:?}", e)));
+                filtered_list.unwrap()
```

Comment:
The `.unwrap` call can be removed and return the error to the caller using `?`

```suggestion
                BlobSidecarList::new(list)
                    .map_err(|e| warp_utils::reject::custom_server_error(format!("{:?}", e)))?
```